File: 1.Domain/Domain.Model/Offer.cs
using Core;
using Domain.Model.Enum;
using System;
using System.Collections.Generic;
using System.Text;

namespace Domain.Model
{
    public class Offer: Entity<int>
    {                
        public DateTime? OfferDate { get; set; }

        public float Salary { get; set; }

        public string RejectionReason { get; set; }   
        
        public OfferStatus Status { get; set; }
    }
}

File: 4.Clients/WebApi/ClientApp/src/entities/offer.ts
import { OfferStatusEnum } from "./enums/offer-status.enum";


export class Offer {
    constructor(id?: number) {
        this.id = id;
      }
    id : number;
    offerDate : Date
    salary : number;
    rejectionReason: string;
    status : OfferStatusEnum
}
